File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/di/module/RepositoryModule.java
package com.kalashnyk.denys.samplemvvmapp.di.module;

import com.kalashnyk.denys.samplemvvmapp.di.scope.RepositoryScope;
import com.kalashnyk.denys.samplemvvmapp.repository.AppRepository;
import com.kalashnyk.denys.samplemvvmapp.repository.database.dao.UserDao;
import com.kalashnyk.denys.samplemvvmapp.repository.server.ServerCommunicator;

import dagger.Module;
import dagger.Provides;

@Module
public class RepositoryModule {
    @RepositoryScope
    @Provides
    AppRepository providesRepository(ServerCommunicator communicator, UserDao productDAO) {
        return new AppRepository(communicator, productDAO);
    }
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/di/module/RepositoryModule.kt
package com.kalashnyk.denys.kotlinsample.di.module

import com.kalashnyk.denys.kotlinsample.di.scope.RepositoryScope
import com.kalashnyk.denys.kotlinsample.repository.AppRepository
import com.kalashnyk.denys.kotlinsample.repository.database.AppDatabase
import com.kalashnyk.denys.kotlinsample.repository.server.ServerCommunicator
import dagger.Module
import dagger.Provides

@Module
class RepositoryModule {
    @RepositoryScope
    @Provides
    internal fun providesRepository(communicator: ServerCommunicator, database: AppDatabase): AppRepository {
        return AppRepository(communicator, database)
    }
}